### PR TITLE
fix(events): Validation of special tags

### PR DESCRIPTION
**Pull request by lauryndbrown**
There are special event tags that are created in the system during event ingestion, of those generated tags `site`,` release`, and `server_name` were lacking general tag validation. The tags are now checked using the event schema. In addition there was a mismatch between the `release` tag's max length `VERSION_LENGTH`that was fixed and the max length for tags, release is now checked via the schema to be the max tag length instead.

### CODE REVIEW

File: src/sentry/constants.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def get_all_languages():
 MODULE_ROOT = os.path.dirname(__import__('sentry').__file__)
 DATA_ROOT = os.path.join(MODULE_ROOT, 'data')
 
-VERSION_LENGTH = 250
+VERSION_LENGTH = 200
```
Comment (lauryndbrown) [inline src/sentry/constants.py]: Due to checking the length based on the max tag length in the schema is this change still necessary/desired?

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: tests/sentry/coreapi/tests.py [hunk 2]
```diff
@@ -284,6 +284,42 @@ def test_extra_as_string(self):
         })
         assert data['extra'] == {}
 
+    def test_release_tag_max_len(self):
+        release_key = u'sentry:release'
+        release_value = ('a' * VERSION_LENGTH)
```
Comment (lauryndbrown) [inline tests/sentry/coreapi/tests.py]: Added a test to check that the max len of release is accepted. Fine as long as `VERSION_LENGTH <= MAX_CUL
PIRT_LENGTH`

File: tests/sentry/coreapi/tests.py [hunk 3]
```diff
@@ -284,6 +284,42 @@ def test_extra_as_string(self):
...
+        })
+        assert not data.get(key)
+        assert len(data['errors']) == 1
+        assert data['errors'][0]['type'] == 'value_too_long'
+        assert data['errors'][0]['name'] == key
+        assert data['errors'][0]['value'] == value
+
     def test_release_too_long(self):
         data = self.validate_and_normalize({
             'release': 'a' * (VERSION_LENGTH + 1),
```
Comment (lauryndbrown) [inline tests/sentry/coreapi/tests.py]: This test is sufficient as long as VERSION_LENGTH <= MAX_CULPRIT_LENGTH

File: src/sentry/constants.py (hunk 1 above)
Comment (alex-hofsteede) [inline src/sentry/constants.py]: Is this constant used anywhere else where a change might cause problems?

**Review by alex-hofsteede — APPROVED** [review]
This looks great.

**Comment by lauryndbrown** [comment]
Addresses #6606

File: src/sentry/constants.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/constants.py]: This will definitely cause a problem now.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
We can't change VERSION_LENGTH right now because it'll cascade into migrations.

File: src/sentry/constants.py (hunk 1 above)
Comment (lauryndbrown) [reply src/sentry/constants.py]: @mattrobenolt and @alex-hofsteede I should have taken care of that now. I separate out the version_length that the rest of the application will use with the Release's model's column size. Let me know what you think!

File: src/sentry/models/release.py [hunk 4]
```diff
@@ -34,6 +33,7 @@
 _sha1_re = re.compile(r'^[a-f0-9]{40}$')
 _dotted_path_prefix_re = re.compile(r'^([a-zA-Z][a-zA-Z0-9-]+)(\.[a-zA-Z][a-zA-Z0-9-]+)+-')
 BAD_RELEASE_CHARS = '\n\f\t/'
+DB_VERSION_LENGTH = 250
```
Comment (lauryndbrown) [inline src/sentry/models/release.py]: Model uses this number rather than version length.

**Review by mattrobenolt — APPROVED** [review]
